src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java

```diff
@@ -53,4 +54,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
         // convert internal representation of user back to API
         return DTOMapper.INSTANCE.convertEntityToUserGetDTO(createdUser);
     }
+
+    @GetMapping("/user/{userId}")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserGetDTO getUserByUserName(@RequestHeader("X-Auth-Token") String token, @PathVariable Long userId){
+        UserGetDTO userGetDTO = new UserGetDTO();
+        return  userGetDTO;
+
+    }
+
+    @GetMapping("/user/login")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @ResponseBody
+    public UserAuthDTO login(@RequestBody UserPostDTO userPostDTO){
+        UserAuthDTO userAuthDTO = new UserAuthDTO();
+        return userAuthDTO;
+    }
+
+    @PutMapping("/logout/{userId}")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void logout(@RequestHeader("X-Auth-Token") String token,@PathVariable Long userId){
+        return;
+    }
+
+    @PutMapping("/user/{userId}/edit")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserGetDTO updateUser(@RequestHeader("X-Auth-Token") String token,@RequestBody User user, @PathVariable Long userId){
```

Comment:
You're using the UserGetDTO object here again for the put request. This is a bit confusing. Can you either implement another object, e.g. UserPutDTO, or use a more generic name if you want to re-use them?

Reply:
Changed the return type to UserUpdateDTO